ARCs/arc-0060.md

```diff
@@ -0,0 +1,330 @@
+
+`SignedDataStr` is the produced 64-byte array, ed25519 digital signature, of the signed data.
+
+```tsx
+export type SignedDataStr = Uint8Array;
+```
+
+#### Enum `ScopeType`
+
+The `ScopeType` enumerates constant strings with the scope of a signing action.
+
+This ARC introduces two scope types.
+
+| ScopeType | Description |
+| --- | --- |
+| MSGSIG | Signature of a simple message. This is the most generic scope. |
+| LSIG | Signature of an Algorand program for delegation. |
+| ... | ... |
+
+Any extension of this ARC **SHOULD** adopt the `ScopeType` above, or introduce a new scope.
+
+#### Interface `StdSignMetadata`
+
+A `StdSignMetadata` object specifies metadata of a `StdSignData` that is being signed.
+
+```tsx
+export interface StdSignMetadata {
+    /**
+    * The scope value of the signing data request.
+    */
+    scope: ScopeType;
+
+    /**
+    * Canonical representation of the JSON schema for the signing data.
+    */
+    schema: string;
+
+    /**
+    * Optional message explaining the reason for the signature.
+    */
+    message?: string;
+
+    /**
+    * Optional encoding used to represent the signing data.
+    */
+    encoding?: string;
+}
+```
+
+If the `encoding` is not specified, the `StdData` object should be UTF-8 encoded following the <a href="https://www.rfc-editor.org/rfc/rfc8785">RFC 8785</a>.
+
+##### Simple Message Data JSON Schema
+
+The JSON schema for simple message signing. The `StdData` **MUST** be an object of this schema.
+
+> The signing data JSON schema is inspired by the schema proposed with <a href="https://eips.ethereum.org/EIPS/eip-712">EIP-712: Typed structured data hashing and signing proposal</a>.
+
+```json
+{
+  "type": "object",
+  "properties": {
+    "ARC60Domain": {
+      "type": "string",
+      "description": "The ARC-60 domain separator"
+    },
+    "bytes": {
+      "type": "string",
+      "description": "Byte string being signed. Used with ScopeType==MSGSIG."
+    },
+  "required": ["ARC60Domain", "bytes"]
+}
+```
+
+- The `ARC60Domain` object indicates the domain separator to be used for signing. It **SHOULD** be set to `"arc60"` for simple bytes. Algorand transaction domain separators `TX` and `TG` **MUST** be forbidden.
+- The `bytes` object is the byte array of data being signed. It is a simple string in case of `ScopeType==MSGSIG`, a more complex schema **SHOULD** be used otherwise. It cannot be a valid Algorand transaction. The value of `bytes` **MUST NOT** be prepended with a known domain separator `TX` or `TG`.
+
+> Algorand domain separators can be found in the <a href="https://github.com/algorandfoundation/specs/blob/master/dev/crypto.md#domain-separation">Algorand specs</a> and in the <a href="https://github.com/algorand/go-algorand/blob/master/protocol/hash.go#L21">Go type HashID</a>.
+
+For example, a valid object that takes as bytes an extra property field is:
+
+```json
+{
+  "ARC60Domain" : "arc60",
+  "bytes" : "ARC-60 is awesome"
+}
+```
```

Comment:
Also here `bytes` might be an integer array?

Reply:
Same as above -> https://github.com/algorandfoundation/ARCs/pull/284/commits/b244ed632ed393a0b33fea1fb0f6c4de03e74506